rails_programming/advanced_forms_and_activerecord/active_record_callbacks.md

```diff
@@ -69,7 +69,7 @@ You can also use conditional logic options `:if` and `:unless` to try a method b
 
 ### Transaction Callbacks
 
-Sometimes your Rails app will need to interact with an external application (which is inherently imperfect) as a part of the save process.  Other times your save will involve juggling several balls at once and, if one fails, they all need to be rolled back.  Typically these cases will involve wrapping your database save operation in a "transaction", which means that either all the steps work or they all fail and are rolled back.
+Sometimes your Rails app will need to interact with an external application (which is inherently imperfect) as a part of the save process.  Other times your save will involve juggling several balls at once and, if one fails, they all need to be rolled back.  Typically these cases will involve wrapping your database save operation in a "transaction," which means that either all the steps work or they all fail and are rolled back.
```

Comment:
should the comma be within the quotes?

Reply:
As far as I understand it, commas do always go inside quotation marks. Here are two relevant references:
https://style.mla.org/punctuation-and-quotation-marks/
https://www.grammarly.com/blog/quotation-marks/
Let me know if you find anything different!

Reply:
ahh TIL, thanks for the explanation @msespos